games/api/views.py

```diff
@@ -105,10 +105,21 @@ class GameSessionBookViewSet(mixins.UpdateModelMixin, viewsets.GenericViewSet):
 
     def update(self, request, *args, **kwargs):
         instance = self.get_object()
+        was_booked = instance.dm is not None
+        print("Was booked", was_booked)
+
         needed_dm = instance.adventure and not instance.dm
         if instance.adventure and instance.dm and instance.dm != request.user.profile:
             return Response(status=status.HTTP_400_BAD_REQUEST)
+
         update_result = super(GameSessionBookViewSet, self).update(request, *args, **kwargs)
+
+        instance.refresh_from_db()
+        print("DM:", instance.dm)
+        print("Is Online:", instance.is_online)
+        if not was_booked and instance.dm is not None and instance.is_online:
```

Comment:
what's the point of having `was_booked and instance.dm is not None` if `was_booked = instance.dm is not None`?

Reply:
mind the `refresh_from_db()` in :116 -- that compares state before and state after the update.

btw. you reminded me I left prints 🤦 gotta remove them asap :D

Reply:
didn't notice that `refresh` because of them XD